### ISSUE TITLE
Converting from closed Paths to [SCNVector3]

### ISSUE THREAD

**Issue by iwoj** [nicklockwood/Euclid#28]
Hey Nick.

I have to make an edit to get closed paths working for me. In my case I was trying to get the [SCNVector3] back out of the close Path and I needed to make this edit to get it to work:

https://github.com/iwoj/Euclid/commit/0d46eea1b43eaa2058fe029672cbad0fbd637bfc

I'm guessing this may break other stuff of yours so I'm not putting in a PR yet. Why do you filter out overlapping vertices?

**Comment by nicklockwood** [comment]
@iwoj the purpose here is not to filter out overlapping vertices, it's to avoid redundancy in the mesh data. Meshes are defined as a set of vertices and a set of indices into those vertices. If multiple vertices have the same coordinate values, you can just duplicate the index instead of the whole vertex.

For example, to describe a quad you need two triangles. If you didn't do this deduplication, that would means storing 6 vertices, but this way you only need 4 vertices.

**Comment by nicklockwood** [comment]
@iwoj I think for the particular use-case of paths, this is probably not necessary since duplication will be minimal. I'd be curious why this is causing a bug though - do you have an example path that isn't rendering correctly with this logic in place?

**Comment by nicklockwood** [comment]
Fixed in 0.3.4

**Comment by nicklockwood** [comment]
@iwoj I think I may have misunderstood what you were trying to do. If you were using this method to build an array of SCNVector then that's not the correct way to do that. It will probably work the way you want in 0.3.4, but it will probably break it again in the future.

It's also not very efficient. The right way to get an array of SCNVector from a path would be to call the `edgeVertices` method directly and then map the positions to vectors yourself, e.g.

```swift
let scnVectors = path.subpaths.flatMap { $0.edgeVertices.map { SCNVector3($0.position) }}
```
